src/realm/sync/noinst/client_reset_operation.cpp

```diff
@@ -67,83 +51,57 @@ bool ClientResetOperation::is_fresh_path(const std::string& path)
     return path.substr(path.size() - suffix_len, suffix_len) == c_fresh_suffix;
 }
 
-bool ClientResetOperation::finalize(sync::SaltedFileIdent salted_file_ident, sync::SubscriptionStore* sub_store,
-                                    util::UniqueFunction<void(int64_t)> on_flx_version_complete)
+bool perform_client_reset(util::Logger& logger, DB& db, DB& fresh_db, ClientResyncMode mode,
+                          CallbackBeforeType notify_before, CallbackAfterType notify_after,
+                          sync::SaltedFileIdent new_file_ident, sync::SubscriptionStore* sub_store,
+                          util::FunctionRef<void(int64_t)> on_flx_version, bool recovery_is_allowed)
 {
-    m_salted_file_ident = salted_file_ident;
+    REALM_ASSERT(mode != ClientResyncMode::Manual);
+    logger.debug("Possibly beginning client reset operation: realm_path = %1, mode = %2, recovery_allowed = %3",
+                 db.get_path(), mode, recovery_is_allowed);
+
+    auto always_try_clean_up = util::make_scope_exit([&]() noexcept {
+        std::string path_to_clean = fresh_db.get_path();
+        try {
+            fresh_db.close();
+            constexpr bool delete_lockfile = true;
+            DB::delete_files(path_to_clean, nullptr, delete_lockfile);
+        }
+        catch (const std::exception& err) {
+            logger.warn("In ClientResetOperation::finalize, the fresh copy '%1' could not be cleaned up due to "
+                        "an exception: '%2'",
+                        path_to_clean, err.what());
+            // ignored, this is just a best effort
+        }
+    });
+
     // only do the reset if there is data to reset
     // if there is nothing in this Realm, then there is nothing to reset and
     // sync should be able to continue as normal
-    auto latest_version = m_db->get_version_id_of_latest_snapshot();
-
-    bool local_realm_exists = latest_version.version != 0;
-    m_logger.debug("ClientResetOperation::finalize, realm_path = %1, local_realm_exists = %2, mode = %3",
-                   m_db->get_path(), local_realm_exists, m_mode);
+    auto latest_version = db.get_version_id_of_latest_snapshot();
+    bool local_realm_exists = latest_version.version > 1;
```

Comment:
shouldn't this check be `>=`?

Reply:
No, the old check was incorrect. The version number for a never-written-to file is 1 and the first write produces version 2.